endpoint.cpp

```diff
@@ -346,35 +349,50 @@ void Endpoint::log_aggregate(unsigned int interval_sec)
     }
 }
 
-int UartEndpoint::open(const char *path, speed_t baudrate)
+UartEndpoint::~UartEndpoint()
+{
+    if (_change_baud_timeout) {
+        Mainloop::get_instance().del_timeout(_change_baud_timeout);
+        _change_baud_timeout = nullptr;
+    }
+}
+
+int UartEndpoint::open(const char *path, unsigned long baudrates[], size_t baud_size)
```

Comment:
Hmmm - so it's caller (in the case, mainloop) task to keep retrying on different bauds? So another callback so main can set speed? Maybe it's too late on Friday and I lost my reasoning skills, but it appears strange.
And your PR has a `set_speed` method: should I rebase this on top of your pr?